library/core/src/fmt/mod.rs

```diff
@@ -94,16 +94,14 @@ pub struct Error;
 
 /// A collection of methods that are required to format a message into a stream.
 ///
-/// This trait is the type which this modules requires when formatting
-/// information. This is similar to the standard library's [`io::Write`] trait,
-/// but it is only intended for use in libcore.
-///
-/// This trait should generally not be implemented by consumers of the standard
-/// library. The [`write!`] macro accepts an instance of [`io::Write`], and the
-/// [`io::Write`] trait is favored over implementing this trait.
-///
-/// [`write!`]: ../../std/macro.write.html
-/// [`io::Write`]: ../../std/io/trait.Write.html
+/// This trait is used by this module for formatting information.
+/// Its capabilities are a subset of the [`std::io::Write`] trait because it only works for
+/// UTF-8–encoded data and it is not [flushable]. If you only want to accept Unicode
```

Comment:
I still feel like this subset is potentially confusing and overall unhelpful. We already say that if you want to accept raw bytes or need flushing then io::Write is appropriate later on; can we just drop this sentence?

Reply:
What about this: 6da6ed9d848?